src/math/util.hpp

```diff
@@ -38,6 +38,12 @@ const T& clamp(const T& val, const T& min, const T& max)
   }
 }
 
+template<class T>
+bool in_bounds(const T& val, const T& min, const T& max)
+{
+  return !(val < min) && (val < max);
```

Comment:
Why not 
```
val > min
```

?

Also, this is inconsistent: min is inclusive while max is exclusive.